osu.Game.Rulesets.Osu/Difficulty/OsuDifficultyCalculator.cs

```diff
@@ -126,13 +130,17 @@ protected override IEnumerable<DifficultyHitObject> CreateDifficultyHitObjects(I
 
         protected override Skill[] CreateSkills(IBeatmap beatmap, Mod[] mods, double clockRate)
         {
-            return new Skill[]
+            var skills = new List<Skill>
             {
                 new Aim(mods, true),
                 new Aim(mods, false),
-                new Speed(mods),
-                new Flashlight(mods)
+                new Speed(mods)
             };
+
+            if (mods.Any(h => h is OsuModFlashlight))
+                skills.Add(new Flashlight(mods));
```

Comment:
This is a bit of a scary one because it requires nobody to attempt to access the "fourth skill" as these calculators will routinely try to do. _As far as I can see_ it should be fine in this case because the only access is guarded with the same flashlight check but yeah.

I really don't like this API. Would hope we can stop playing the index game here at some point and just use straight objects or something. At least a dict.

Reply:
I was thinking about using an object until I realised that it's an overriden method so would require consideration for that. I agree with you entirely though, perhaps a separate PR I can create to refactor.

Reply:
Something for later.